### DISCUSSION TITLE
Convert a pytorch torch.nn custom module class to a diffusers style model

### DISCUSSION THREAD

**Discussion by nighting0le01** [huggingface/diffusers#7547] [Q&A]
Hi how can i convert my custom pure torch.nn.module model to be similar to diffusers model. I want to basically add my custom VAE and use with training a regular VAE.

**Comment by sayakpaul** [comment]
I don't think there will be much. You can follow this PR to get a rough understanding of what needs to be done:

https://github.com/huggingface/diffusers/pull/4384

**Reply by nighting0le01** [reply]
Thanks sayak, can you point me to some relevant documentation for this and also on how to incorporate my VAE into a custom pipeline? I would like to have 2  VAE Encoder and 2 VAE Decoder in the pipeline in the end. the VAE decoder is actually a Unet.

**Reply by sayakpaul** [reply]
This should be helpful: https://huggingface.co/docs/diffusers/main/en/using-diffusers/custom_pipeline_overview.
